### PR TITLE
[POSTPONED] cache coherence: fix bugs and enable CONFIG_INCOHERENT

### PR DESCRIPTION
**Pull request by lyakh**
CONFIG_INCOHERENT was introduced a while ago but was never actually enabled. Enabling it now breaks basic SOF functionality. The first of these three patches prepares for the fix by removing `comp_make_shared()`, the second one fixes bugs that cause those breakages and the third one enables CONFIG_COHERENT. This PR doesn't depend on any other pending PRs and can be merged independently
UPDATE 16.05: this is postponed until other buffer acquisition fixes are merged

### CODE REVIEW

**Comment by lgirdwood** [comment]
@wszypelt can you help here with the failed IPC4 logs ? Thanks.

**Comment by lgirdwood** [comment]
> @lgirdwood turns out this isn't enough. This builds but doesn't work without more patches from the whole set. I could pull those together into this PR, but maybe we just leave it as is until the rest is merged, then this should work too.

ok, so are the commits bisectable ?

**Comment by wszypelt** [comment]
@lgirdwood from internal logs i can only see:
for both tests
```
15:56:29,473 INFO  - cavs_scripts-py\utilities\test_logger.py:709: in wrapper
15:56:29,473 INFO  -     returned = method(self, *args, **kwargs)
15:56:29,473 INFO  - cavs_scripts-py\utilities\ipc_driver.py:820: in send
15:56:29,473 INFO  -     read_response_payload=message.READ_RESPONSE_PAYLOAD)
15:56:29,473 INFO  - cavs_scripts-py\utilities\test_logger.py:709: in wrapper
15:56:29,473 INFO  -     returned = method(self, *args, **kwargs)
15:56:29,473 INFO  - cavs_scripts-py\utilities\ipc_driver.py:652: in block_on_response
15:56:29,473 INFO  -     expect_error=expect_error, expected_reply_size=expected_reply_size)
15:56:29,474 INFO  - cavs_scripts-py\utilities\ipc_driver.py:770: in _block_on_response_read_avs
15:56:29,474 INFO  -     rec_ipc = self.read(exit_right_away=True, expect_error=expect_error, expected_reply_size=expected_reply_size)
15:56:29,474 INFO  - cavs_scripts-py\utilities\ipc_driver.py:462: in read
15:56:29,474 INFO  -     self.raise_or_save(error)
15:56:29,474 INFO  - cavs_scripts-py\utilities\ipc_driver.py:632: in raise_or_save
15:56:29,474 INFO  -     raise error
15:56:29,474 INFO  - E   utilities.ipc_driver_exceptions.AvsIpcErrorResponseException: FW responded with error
15:56:29,474 INFO  - E   FW status:                    0x00000005
15:56:29,474 INFO  - E   IPC error:                             7 (ADSP_IPC_INVALID_REQUEST)
15:56:29,474 INFO  - ---------------------------- Captured stdout setup -----------------------------
```

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
IPC4 fixes merged recently so retest.

**Comment by lgirdwood** [comment]
@lyakh can you rebase and push. Thanks.

**Comment by XiaoyunWu6666** [comment]
SOFCI TEST

**Comment by XiaoyunWu6666** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@wszypelt do we still have the same error log as before ?
@RanderWang can you help here as I'd like to get this in for v2.2 @mengdonglin fyi.

**Comment by lgirdwood** [comment]
@wszypelt is it possible to attach a bigger log so debug team can see the whole flow that fails. Thanks!

**Comment by wszypelt** [comment]
@lgirdwood full log from 01_02_copiers[enable_power_gating=False-16000Hz_32in32bit_2ch
[full-log (1).txt](https://github.com/thesofproject/sof/files/8686898/full-log.1.txt)
]

**Comment by lyakh** [comment]
> > @lgirdwood turns out this isn't enough. This builds but doesn't work without more patches from the whole set. I could pull those together into this PR, but maybe we just leave it as is until the rest is merged, then this should work too.
> 
> ok, so are the commits bisectable ?

@lgirdwood yes, sure

**Comment by lyakh** [comment]
> @lyakh can you rebase and push. Thanks.

@lgirdwood let's wait with this until we merge other component-specific fixes, beginning with #5801

**Comment by RanderWang** [comment]
> @wszypelt do we still have the same error log as before ? @RanderWang can you help here as I'd like to get this in for v2.2 @mengdonglin fyi.

It passed my IPC4 test on TGL.

File: src/ipc/ipc3/helper.c [hunk 1]
```diff
@@ -566,7 +566,7 @@ static int ipc_comp_to_buffer_connect(struct ipc_comp_dev *comp,
 		coherent_shared(buffer->cb, c);
 
 		if (!comp->cd->is_shared)
-			comp->cd = comp_make_shared(comp->cd);
+			comp_make_shared(comp->cd);
```
Comment (RanderWang) [inline src/ipc/ipc3/helper.c:569]: can we convert 
````c
if (!comp->cd->is_shared)
    comp_make_shared(comp->cd); 
````
to
````c
comp_make_shared(comp->cd); 
````

**Comment by lyakh** [comment]
superseded and mostly merged

**Pull request closed without merging**